mysql-test/suite/versioning/t/trx_id.test

```diff
@@ -411,3 +411,154 @@ DROP TABLE tts;
+
+--echo with table rebuild
+SELECT c.prtype FROM INFORMATION_SCHEMA.INNODB_SYS_COLUMNS AS c
+  INNER JOIN INFORMATION_SCHEMA.INNODB_SYS_TABLES AS t
+    ON c.table_id=t.table_id
+  WHERE t.name='test/t' AND c.name='a';
+--enable_info
+ALTER TABLE t
+  CHANGE a a INT WITH SYSTEM VERSIONING,
+  ADD PRIMARY KEY pk(a);
+--disable_info
+SELECT c.prtype FROM INFORMATION_SCHEMA.INNODB_SYS_COLUMNS AS c
+  INNER JOIN INFORMATION_SCHEMA.INNODB_SYS_TABLES AS t
+    ON c.table_id=t.table_id
+  WHERE t.name='test/t' AND c.name='a';
+
+UPDATE t SET a=1;
+SELECT COUNT(*) FROM t FOR SYSTEM_TIME ALL;
+
+SHOW CREATE TABLE t;
+
+-- echo handles VIRTUAL columns too
+CREATE OR REPLACE TABLE t (
+  a INT AS (b + 1),
+  b INT,
+  row_start BIGINT UNSIGNED AS ROW START INVISIBLE,
+  row_end BIGINT UNSIGNED AS ROW END INVISIBLE,
+  PERIOD FOR SYSTEM_TIME(row_start, row_end)
+) WITH SYSTEM VERSIONING ENGINE=INNODB;
+
+INSERT INTO t VALUES (DEFAULT, 1);
+
+SELECT c.prtype FROM INFORMATION_SCHEMA.INNODB_SYS_COLUMNS AS c
+  INNER JOIN INFORMATION_SCHEMA.INNODB_SYS_TABLES AS t
+    ON c.table_id=t.table_id
+  WHERE t.name='test/t' AND c.name='b';
+--enable_info
+ALTER TABLE t
+  CHANGE b b INT WITHOUT SYSTEM VERSIONING;
+--disable_info
+SELECT c.prtype FROM INFORMATION_SCHEMA.INNODB_SYS_COLUMNS AS c
+  INNER JOIN INFORMATION_SCHEMA.INNODB_SYS_TABLES AS t
+    ON c.table_id=t.table_id
+  WHERE t.name='test/t' AND c.name='b';
+
+UPDATE t SET b=11;
+SELECT COUNT(*) FROM t FOR SYSTEM_TIME ALL;
+
+
+--echo check attribute persistence
+CREATE OR REPLACE TABLE t (
+  a INT,
+  b INT,
+  row_start BIGINT UNSIGNED AS ROW START INVISIBLE,
+  row_end BIGINT UNSIGNED AS ROW END INVISIBLE,
+  PERIOD FOR SYSTEM_TIME(row_start, row_end)
+) WITH SYSTEM VERSIONING ENGINE=INNODB;
+
+INSERT INTO t VALUES (1,1);
+
+--echo without rebuild
+--enable_info
+ALTER TABLE t
+  CHANGE a a INT WITHOUT SYSTEM VERSIONING;
+--disable_info
+
+--source include/shutdown_mysqld.inc
+--source include/start_mysqld.inc
+
+UPDATE t SET a=2;
+SELECT COUNT(*) FROM t FOR SYSTEM_TIME ALL;
+
+--echo with rebuild
+SET @@SYSTEM_VERSIONING_ALTER_HISTORY=KEEP;
+--enable_info
+ALTER TABLE t
+  CHANGE a a INT WITH SYSTEM VERSIONING,
+  ADD PRIMARY KEY pk (a);
+--disable_info
+
+--source include/shutdown_mysqld.inc
+--source include/start_mysqld.inc
+
+UPDATE t SET a=3;
+SELECT COUNT(*) FROM t FOR SYSTEM_TIME ALL;
+
+--echo rollback ALTER TABLE: nothing should change
+SET @@SYSTEM_VERSIONING_ALTER_HISTORY=KEEP;
+CREATE OR REPLACE TABLE t (
+  a INT,
+  b INT,
+  row_start BIGINT UNSIGNED AS ROW START INVISIBLE,
+  row_end BIGINT UNSIGNED AS ROW END INVISIBLE,
+  PERIOD FOR SYSTEM_TIME(row_start, row_end)
+) WITH SYSTEM VERSIONING ENGINE=INNODB;
+
+INSERT INTO t VALUES (1, 1);
+
+SELECT C.PRTYPE FROM INFORMATION_SCHEMA.INNODB_SYS_COLUMNS AS C
+  JOIN INFORMATION_SCHEMA.INNODB_SYS_TABLES AS t ON C.TABLE_ID=t.TABLE_ID
+  WHERE t.NAME='test/t' AND C.NAME='b';
+
+SET DEBUG_DBUG='+d,ib_commit_inplace_fail_1';
```

Comment:
Before this, I would do
```sql
SET @saved_debug_dbug = @@SESSION.debug_dbug;
```
and after the instrumented statement, restore the setting:
```sql
SET DEBUG_DBUG = @saved_debug_dbug;
```
In this way, if further tests are added to the same file, there will be no surprises due to this fault injection still being active.